timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/AllNodesDisabledNamespacesUpdater.java

```diff
@@ -0,0 +1,271 @@
+    private final ImmutableList<DisabledNamespacesUpdaterService> remoteUpdaters;
+    private final Map<DisabledNamespacesUpdaterService, CheckedRejectionExecutorService> remoteExecutors;
+    private final TimelockNamespaces localUpdater;
+    private final Supplier<UUID> lockIdSupplier;
+    private final int clusterSize;
+
+    @VisibleForTesting
+    AllNodesDisabledNamespacesUpdater(
+            AuthHeader authHeader,
+            ImmutableList<DisabledNamespacesUpdaterService> remoteUpdaters,
+            Map<DisabledNamespacesUpdaterService, CheckedRejectionExecutorService> remoteExecutors,
+            TimelockNamespaces localUpdater,
+            Supplier<UUID> lockIdSupplier) {
+        this.authHeader = authHeader;
+        this.remoteUpdaters = remoteUpdaters;
+        this.remoteExecutors = remoteExecutors;
+        this.localUpdater = localUpdater;
+        this.lockIdSupplier = lockIdSupplier;
+        this.clusterSize = remoteUpdaters.size() + 1;
+    }
+
+    public static AllNodesDisabledNamespacesUpdater create(
+            AuthHeader authHeader,
+            ImmutableList<DisabledNamespacesUpdaterService> updaters,
+            Map<DisabledNamespacesUpdaterService, CheckedRejectionExecutorService> executors,
+            TimelockNamespaces localUpdater) {
+        return new AllNodesDisabledNamespacesUpdater(authHeader, updaters, executors, localUpdater, UUID::randomUUID);
+    }
+
+    public DisableNamespacesResponse disableOnAllNodes(Set<Namespace> namespaces) {
+        if (anyNodeIsUnreachable()) {
+            return DisableNamespaceResponses.unsuccessfulDueToPingFailure(namespaces);
+        }
+
+        UUID lockId = lockIdSupplier.get();
+        List<SingleNodeUpdateResponse> responses = disableNamespacesOnAllNodes(namespaces, lockId);
+        if (updateWasSuccessfulOnAllNodes(responses)) {
+            return DisableNamespacesResponse.successful(SuccessfulDisableNamespacesResponse.of(lockId));
+        }
+
+        Set<Namespace> consistentlyLockedNamespaces = getConsistentFailures(responses);
+        if (!consistentlyLockedNamespaces.isEmpty()) {
+            return DisableNamespaceResponses.unsuccessfulDueToConsistentlyLockedNamespaces(
+                    consistentlyLockedNamespaces);
+        }
+
+        // If no namespaces were consistently disabled, we should roll back our request,
+        // to avoid leaving ourselves in an inconsistent state.
+        boolean rollbackSuccess = attemptReEnableOnAllNodes(namespaces, lockId, responses.size());
+        if (rollbackSuccess) {
+            return DisableNamespaceResponses.unsuccessfulButRolledBack(namespaces, lockId);
+        }
+
+        return DisableNamespaceResponses.unsuccessfulAndRollBackFailed(namespaces, lockId);
+    }
+
+    public ReenableNamespacesResponse reEnableOnAllNodes(ReenableNamespacesRequest request) {
+        Set<Namespace> namespaces = request.getNamespaces();
+
+        if (anyNodeIsUnreachable()) {
```

Comment:
We can probably skip this ping -- yes we may partially unlock, but that's no worse than this failing, if we do the unlocking best effort